app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/db/DatabaseInitUtil.java

```diff
@@ -0,0 +1,185 @@
+import java.util.List;
+
+public class DatabaseInitUtil {
+    // For debugging purpose
+    private static final boolean DEBUG = false;
+    private static final String TAG = "VideoDbBuilder";
+
+    public void initializeDb(AppDatabase db, String url) throws IOException{
+
+        // json data
+        String json;
+
+        if (AppConfiguration.IS_DEBUGGING_VERSION) {
+
+            // when use debugging version, we won't fetch data from network but using local
+            // json file (only contain 4 video entities in 2 categories.)
+            json = Utils.inputStreamToString(SampleApplication
+                    .getInstance()
+                    .getApplicationContext()
+                    .getResources()
+                    .openRawResource(R.raw.live_movie_debug));
+        } else {
+            // Fetch json data from network
+            json = fetchJsonStringFromUrl(url);
+        }
+        buildDatabase(json, db);
+    }
+
+    /**
+     * Takes the contents of a JSON object and populates the database
+     *
+     * @param jsonString The JSON String of all videos' information
+     */
+    public static void buildDatabase(String jsonString, AppDatabase db) throws IOException {
+        Gson gson = new Gson();
+        VideosWithGoogleTag videosWithGoogleTag = gson.fromJson(jsonString,
+                VideosWithGoogleTag.class);
+        for (VideosGroupByCategory videosGroupByCategory: videosWithGoogleTag.getAllResources()) {
+
+            // create category table
+            CategoryEntity categoryEntity = new CategoryEntity();
+            categoryEntity.setCategoryName(videosGroupByCategory.getCategory());
+            db.categoryDao().insertCategory(categoryEntity);
+
+            // create video table
+            postProcessing(videosGroupByCategory);
+            db.videoDao().insertAllVideos(videosGroupByCategory.getVideos());
+        }
+    }
+
+    /**
+     * Helper function to make some customization on raw data
+     */
+    private static void postProcessing(VideosGroupByCategory videosGroupByCategory) {
+        for (VideoEntity each: videosGroupByCategory.getVideos()) {
+            each.setCategory(videosGroupByCategory.getCategory());
+            each.setVideoLocalStorageUrl("");
+            each.setVideoBgImageLocalStorageUrl("");
+            each.setVideoCardImageLocalStorageUrl("");
+            each.setVideoUrl(each.getVideoUrls().get(0));
+            each.setRented(false);
+            each.setStatus("");
+            each.setTrailerVideoUrl("https://storage.googleapis.com/android-tv/Sample%20videos/Google%2B/Google%2B_%20Say%20more%20with%20Hangouts.mp4");
+        }
+    }
+
+    /**
+     * Fetch JSON string from a given URL.
+     *
+     * @return the JSONObject representation of the response
+     * @throws IOException The BufferedReader operation may throw an IOException.
+     */
+    private static String fetchJsonStringFromUrl(String urlString) throws IOException {
```

Comment:
use a library instead of doing this manually. Retrofit is recommended. you are not handling things like redirects and it is bad to re-invent the wheel.

Reply:
Thank you for your great suggestion sir! The library is awesome, much  boilerplate code can be removed with it.